OsmAnd/src/net/osmand/plus/routing/RoutingHelper.java

```diff
@@ -265,6 +267,33 @@ public void addListener(IRouteInformationListener l){
 		transportRoutingHelper.addListener(l);
 	}
 
+	public void addDataUpdateListener(IRoutingDataUpdateListener listener) {
+		List<WeakReference<IRoutingDataUpdateListener>> copyList = new ArrayList<>(updateListeners);
+		Iterator<WeakReference<IRoutingDataUpdateListener>> it = copyList.iterator();
+		while (it.hasNext()) {
```

Comment:
Looks very similar to removeDataUpdateListener, 5-7 lines of code could be moved to a common private method

Reply:
+